src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1879,7 +1879,9 @@ def add_vm_secret(cmd, resource_group_name, vm_name, keyvault, certificate, cert
 
 def list_vm_secrets(cmd, resource_group_name, vm_name):
     vm = get_vm(cmd, resource_group_name, vm_name)
-    return vm.os_profile.secrets
+    if vm.os_profile:
+        return vm.os_profile.secrets
+    return None
```

Comment:
`return None` will show nothing in the output.
how about `return []`

Reply:
Nice. actually I have same confusion on return value when result is None. sometimes az cli returns  [], sometimes None. Is there any general guideline?

Reply:
In this command, the output is [] if there is no secret.
Let me search the doc.

Reply:
Tested with a VM and the results confirms @qwordy's comment:

```
az vm secret list -n {} -g {}
[]
```